### PR TITLE
feat(contracts): add PrizeDistributionSource

### PR DESCRIPTION
**Pull request by PierrickGT**

### CODE REVIEW

File: contracts/interfaces/IDrawCalculator.sol [hunk 1]
```diff
@@ -51,7 +52,7 @@ interface IDrawCalculator {
      * @notice Read global DrawBuffer variable.
      * @return IDrawBuffer
      */
-    function getPrizeDistributionBuffer() external view returns (IPrizeDistributionBuffer);
+    function getPrizeDistributionSource() external view returns (IPrizeDistributionSplitter);
```
Comment (asselstine) [inline contracts/interfaces/IDrawCalculator.sol:55]: Why does the Draw Calculator know about the splitter?

File: contracts/interfaces/IPrizeDistributionSplitter.sol [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+pragma solidity 0.8.6;
+
+import "./IPrizeDistributionBuffer.sol";
+
+/** @title IPrizeDistributionSplitter
+ * @author PoolTogether Inc Team
+ * @notice The PrizeDistributionSplitter interface.
+ */
+interface IPrizeDistributionSplitter {
```
Comment (asselstine) [inline contracts/interfaces/IPrizeDistributionSplitter.sol:11]: IPrizeDistributionSource. ..... ?

**Review by asselstine — CHANGES_REQUESTED** [review]

File: contracts/interfaces/IDrawCalculator.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/interfaces/IDrawCalculator.sol:55]: Cause we need the `getPrizeDistributions` function from the interface.

File: contracts/DrawCalculator.sol [hunk 3]
```diff
@@ -27,7 +27,7 @@ contract DrawCalculator is IDrawCalculator {
     ITicket public immutable ticket;
 
     /// @notice The stored history of draw settings.  Stored as ring buffer.
-    IPrizeDistributionSplitter public immutable prizeDistributionSplitter;
+    IPrizeDistributionSource public immutable prizeDistributionSource;
```
Comment (asselstine) [inline contracts/DrawCalculator.sol:30]: This shouldn't be public.  There's already a getter.

**Review by asselstine — CHANGES_REQUESTED** [review]

File: contracts/DrawCalculator.sol (hunk 3 above)
Comment (PierrickGT) [reply contracts/DrawCalculator.sol]: Done: https://github.com/pooltogether/v4-core/pull/256/commits/7799c597adfb01be2de66bbed4427d15e9ed68c5

**Review by asselstine — APPROVED** [review]

**Comment by kamescg** [comment]
image

Do we want to squash linting errors before merging?

File: contracts/DrawCalculator.sol [hunk 4]
```diff
@@ -9,7 +9,7 @@ import "./interfaces/IPrizeDistributionBuffer.sol";
 import "./interfaces/IDrawBeacon.sol";
 
 /**
-  * @title  PoolTogether V4 DrawCalculator
+  * @title  PoolTogether V4 DrawCalculatorV2
```
[the remark this answers is not in the document]
Comment (PierrickGT) [reply contracts/DrawCalculator.sol]: Yes, wrong contract, I've removed it.

File: contracts/interfaces/IDrawCalculatorV2.sol [hunk 5]
```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-3.0
```
[the remark this answers is not in the document]
Comment (PierrickGT) [reply contracts/interfaces/IDrawCalculatorV2.sol]: I've removed the interface.

**Review by kamescg — APPROVED** [review]

**Pull request merged**